app.py

```diff
@@ -65,6 +66,54 @@
 model_data = JobShopData()
 
 
+@app.callback(
+    Output("left-column", "className"),
+    inputs=[
+        Input("left-column-collapse", "n_clicks"),
+        State("left-column", "className"),
+    ],
+    prevent_initial_call=True,
+)
+def toggle_left_column(left_column_collapse: int, class_name: str) -> str:
+    """Toggles left column 'collapsed' class that hides and shows the left column.
+
+    Args:
+        left_column_collapse (int): The (total) number of times the collapse button has been clicked.
+        class_name (str): Current class name of the left column, 'collapsed' if not visible, empty string if visible
+
+    Returns:
+        str: The new class name of the left column.
+    """
+    if class_name:
+        return ""
+    return "collapsed"
+
+
+@app.callback(
+    Output('solver-select', 'className'),
+    Output('solver-select', 'value'),
+    inputs=[
+        Input('model-select', 'value'),
+    ],
+    prevent_initial_call=True
+)
+def update_solver_options(model_value: str) -> list[str]:
+    """Hides and shows classical solver option using 'hide-classic' class
+
+    Args:
+        model_value (str): Currently selected model from model-select dropdown.
+
+    Returns:
+        str: The new class name of the solver-select checklist.
+        list: Unselects MIP and selects Hybrid or no update.
+    """
+
+    if model_value == 'QM':
+        return 'hide-classic', ['Hybrid']
+    return '', dash.no_update
```

Comment:
What do you think about storing the last setting for the checkboxes when returning to MIP?
```python
    return "hide-classic", ["Hybrid"], last_model_select
return "", last_model_select, dash.no_update
```
and a `dcc.Store("last_model_select)` in the html + another output and input.

Reply:
That's a good idea, it would probably be more intuitive

Reply:
updated: https://github.com/dwave-examples/job-shop-scheduling-cqm/pull/16/commits/70a0bc7b924d93d3a5eec9ca0afe31869ee8bac3